search-setup/__tests__/search.test.ts

```diff
@@ -0,0 +1,51 @@
+import { config } from "dotenv";
+import { DocumentResponse, SearchIndexResponse } from "../types";
+
+config();
+
+const searchEndpoint = process.env["AI_SEARCH_SERVICE_ENDPOINT"] ?? "";
+const indexName = process.env["AI_SEARCH_INDEX_NAME"];
+const apiKey = process.env["AI_SEARCH_API_KEY"] ?? "";
+
+describe("AI search index creation and data loading", () => {
+  const URL_PREFIX = `${searchEndpoint}/indexes('${indexName}')`;
+  const URL_SUFFIX = "?api-version=2024-07-01";
+
+  test("should create index with expected fields", async () => {
+    const url = `${URL_PREFIX}${URL_SUFFIX}`;
+
+    const response = await fetch(url, {
+      headers: {
+        "api-key": apiKey,
+      },
+    });
+
+    const index: SearchIndexResponse = await response.json();
+
+    expect(index.name).toBe(indexName);
+    expect(index.fields.length).toBe(2);
+    expect(index.fields.at(0)?.name).toBe("IID");
+    expect(index.fields.at(0)?.type).toBe("Edm.String");
+    expect(index.fields.at(0)?.key).toBe(true);
+    expect(index.fields.at(0)?.retrievable).toBe(true);
```

Comment:
I think we want to be checking for the value of the `searchable` field here, as it's important to being able to actually search on certain fields.